xrviz/fields.py

```diff
@@ -134,3 +145,20 @@ def set_coords(self, data, var):
 
     def ndim_matches(self, var1, var2):
         return self.data[var1].ndim == self.data[var2].ndim
+
+    def check_are_var_coords(self):
+        '''
+        Check if both x and y are in variable's coords
+        '''
+        var_coords = list(self.data[self.var].coords)
+        x = self.x.value
+        y = self.y.value
+        return True if x in var_coords and y in var_coords else False
+
+    def guess_x_y(self, var):
+        try:
+            parsed_var = self.data.metpy.parse_cf(var)
+            x, y = parsed_var.metpy.coordinates('x', 'y')
+            return [coord.name for coord in (x, y)]
+        except:
```

Comment:
When does this fail?

Reply:
This fails when:
1. `lat`,`lon` are not available as coordinates.
2. For example dataset: when `mask` is also set as coordinate.

I have added a comment.